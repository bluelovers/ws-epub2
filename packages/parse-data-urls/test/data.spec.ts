/**
 * Created by user on 2020/3/31.
 */

import parseDataURL from '../index';
import { readFile, outputFile } from 'fs-extra';
import { join } from 'path';

describe('base64', function ()
{

	it('should decode', async () => {

		const url = "data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAUAAAAFCAYAAACNbyblAAAAHElEQVQI12P4//8/w38GIAXDIBKE0DHxgljNBAAO9TXL0Y4OHwAAAABJRU5ErkJggg==";

		const data = await parseDataURL(url);

		expect(data).toMatchObject({
			type: 'image',
			mime: 'image/png',
			base64: true,
		});
		expect(data).toMatchSnapshot();

		await outputFile(join(__dirname, 'temp', 't001.png'), data.body);

	});

	it('utf-8', async () => {

		const url = "data:image/svg+xml;charset=UTF-8,some-data";

		const data = await parseDataURL(url);

		expect(data).toMatchObject({
			type: 'image',
			mime: 'image/svg+xml',
			base64: false,
			charset: 'UTF-8',
		});
		expect(data).toMatchSnapshot();

	});

});
