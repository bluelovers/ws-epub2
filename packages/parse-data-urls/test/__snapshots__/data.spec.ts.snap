// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`base64 should decode 1`] = `
Object {
  "base64": true,
  "body": Object {
    "data": Array [
      137,
      80,
      78,
      71,
      13,
      10,
      26,
      10,
      0,
      0,
      0,
      13,
      73,
      72,
      68,
      82,
      0,
      0,
      0,
      5,
      0,
      0,
      0,
      5,
      8,
      6,
      0,
      0,
      0,
      141,
      111,
      38,
      229,
      0,
      0,
      0,
      28,
      73,
      68,
      65,
      84,
      8,
      215,
      99,
      248,
      255,
      255,
      63,
      195,
      127,
      6,
      32,
      5,
      195,
      32,
      18,
      132,
      208,
      49,
      241,
      130,
      88,
      205,
      4,
      0,
      14,
      245,
      53,
      203,
      209,
      142,
      14,
      31,
      0,
      0,
      0,
      0,
      73,
      69,
      78,
      68,
      174,
      66,
      96,
      130,
    ],
    "type": "Buffer",
  },
  "charset": undefined,
  "mime": "image/png",
  "mimeType": MIMEType {
    "_parameters": MIMETypeParameters {
      "_map": Map {},
    },
    "_subtype": "png",
    "_type": "image",
  },
  "type": "image",
}
`;

exports[`base64 utf-8 1`] = `
Object {
  "base64": false,
  "body": Object {
    "data": Array [
      115,
      111,
      109,
      101,
      45,
      100,
      97,
      116,
      97,
    ],
    "type": "Buffer",
  },
  "charset": "UTF-8",
  "mime": "image/svg+xml",
  "mimeType": MIMEType {
    "_parameters": MIMETypeParameters {
      "_map": Map {
        "charset" => "UTF-8",
      },
    },
    "_subtype": "svg+xml",
    "_type": "image",
  },
  "type": "image",
}
`;
